components/script/textinput.rs

```diff
@@ -442,12 +442,24 @@ impl<T: ClipboardProvider> TextInput<T> {
         let target_line: isize = self.edit_point.line as isize + adjust;
 
         if target_line < 0 {
-            self.edit_point.index = 0;
             self.edit_point.line = 0;
+            self.edit_point.index = 0;
+            if let Some(_) = self.selection_origin {
+                if self.selection_direction == SelectionDirection::None ||
+                    self.selection_direction == SelectionDirection::Forward
+                {
+                    self.selection_origin = Some(TextPoint { line: 0, index: 0 })
+                }
+            }
             return;
         } else if target_line as usize >= self.lines.len() {
             self.edit_point.line = self.lines.len() - 1;
             self.edit_point.index = self.current_line_length();
+            if let Some(_) = self.selection_origin {
+                if self.selection_direction == SelectionDirection::Backward {
+                    self.selection_origin = Some(self.edit_point)
```

Comment:
Missing semicolon ? but it seems `test-tidy` didn't complain about it 😂

Reply:
Since the assignment has the `unit` type you don't need a semicolon here, but I put it all the same if it's the coding style.